### PR TITLE
[ISSUE-#6763] Make refreshInvoker synchronized

### PR DESCRIPTION
**Pull request by horizonzy**
## What is the purpose of the change
For-#6763
Make refreshInvoker synchronized, to avoid config-center change and registry-center change concurrency.

### CODE REVIEW

**Comment by horizonzy** [comment]
@AlbumenJ This pr also fix concurency bug, PTAL.

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryDirectory.java [hunk 1]
```diff
@@ -152,7 +152,9 @@ private String judgeCategory(URL url) {
         return "";
     }
 
-    private void refreshOverrideAndInvoker(List<URL> urls) {
+    // Issue: https://github.com/apache/dubbo/issues/6763.
```
Comment (AlbumenJ) [inline dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryDirectory.java:155]: remove this line, and we can refer these changes in git blame

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryDirectory.java (hunk 1 above)
Comment (horizonzy) [reply dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryDirectory.java]: done.

**Review by AlbumenJ — APPROVED** [review]
LGTM.

**Pull request merged**